### PR TITLE
Refactored the buyers.

### LINKED ISSUE #43: Refactor the buyers.

### PR DESCRIPTION
**Pull request by StefanBoodt**
I introduced some tests for the buyers. And made the methods smaller and more readable. This branch should fix #43. I personally did not test if the buyers worked online, but I did not alter the code in a way that should destroy the effectiveness of the code working.

I think we should test that anyway but we could do that in combination with testing #40.

### CODE REVIEW

File: agent/OffshoredediBuyer.py [hunk 1]
```diff
@@ -21,25 +21,14 @@ class OffshoredediBuyer(VPSBuyer):
     """
     This class orders a VPS from offshorededi.com
     """
-    def __init__(self, email = "", password = ""):
+    def __init__(self, email="", password=""):
```
Comment (i-am-xhy) [inline agent/OffshoredediBuyer.py]: isn't the style with spaces better? because it reads more normally?

Comment (StefanBoodt) [inline agent/OffshoredediBuyer.py]: I agree but python styling for some reason does not. Look at https://docs.python.org/3/tutorial/controlflow.html#default-argument-values or http://www.diveintopython.net/power_of_introspection/optional_arguments.html for examples.

**Comment by i-am-xhy** [comment]
currently has merge conflicts, furthermore can't be merged yet because offshorededi is currently partly offline

**Comment by StefanBoodt** [comment]
Offshorededibuyer should work now. I came to the pay page (and failed (probably because I did not have any bitcoins)) without error messages.

**Pull request merged**